.github/workflows/ci.yml

```diff
@@ -19,172 +19,318 @@ jobs:
       - run: git remote add upstream https://github.com/modin-project/modin.git
       - run: git fetch upstream
       - run: npx commitlint --from upstream/master --to HEAD --verbose
+
   lint-black:
     name: lint (black)
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: "3.7.x"
           architecture: "x64"
       - run: pip install black
       - run: black --check --diff modin/
+
   lint-flake8:
     name: lint (flake8)
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: "3.7.x"
           architecture: "x64"
       - run: pip install flake8 flake8-print
       - run: flake8 --enable=T modin
+
   test-api:
     runs-on: ubuntu-latest
     name: test api
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 1
-      - uses: actions/setup-python@v1
+      - name: Cache conda
+        uses: actions/cache@v1
         with:
-          python-version: "3.7.x"
-          architecture: "x64"
+          path: ~/conda_pkgs_dir
+          # github.run_id using for create new cache entry for each run
+          # this allows us to use actual packages
+          key: ${{ runner.os }}-python-3.7-conda-${{ github.run_id }}-${{ hashFiles('environment.yml') }}
+      - name: Cache pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-python-3.7-pip-${{ github.run_id }}-${{ hashFiles('environment.yml') }}
+      - uses: goanpeca/setup-miniconda@v1.6.0
+        with:
+          activate-environment: modin
+          environment-file: environment.yml
+          python-version: "3.7"
+          channel-priority: strict
+          use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
+          auto-update-conda: true
+      - name: Conda environment
+        # `shell: bash -l {0}` - special way to activate modin environment
+        shell: bash -l {0}
+        run: |
+          conda info
+          conda list
       - run: sudo apt update && sudo apt install -y libhdf5-dev
-      - run: pip install -r requirements.txt
-      - run: python -m pytest modin/pandas/test/test_api.py
+      - name: Api tests
+        shell: bash -l {0}
```

Comment:
why do we need `bash -l` here?

Reply:
It's come from `https://github.com/goanpeca/setup-miniconda#important`.